Userland/Libraries/LibJS/Lexer.cpp

```diff
@@ -369,8 +370,22 @@ bool Lexer::is_whitespace() const
     return false;
 }
 
-bool Lexer::is_identifier_start() const
+bool Lexer::is_unicode_escape(size_t& identifier_length) const
 {
+    GenericLexer lexer(source().substring_view(m_position - 1));
+    if (lexer.consume_escaped_code_point(false).is_error())
+        return false;
+
+    identifier_length = lexer.tell();
+    return true;
+}
+
+bool Lexer::is_identifier_start(size_t& identifier_length) const
+{
+    identifier_length = 1;
+
+    if (m_current_char == '\\')
+        return is_unicode_escape(identifier_length);
```

Comment:
So here we would need to verify that it would pass the checks below `is_ascii_alpha(...) || == '_'`, etc

Reply:
Do we? At this point we know the next six chars are `\unnnn`, where `n` is a hex digit, so also doing the check below seems like extra work.

Reply:
Oh, or are you saying the resolved code point needs to pass that test? I see

Reply:
Yeah so you cannot get an identifier starting with `0` for example `var \u{30};`